DomainContent/BongoBingo/bingoWheel.js

```diff
@@ -0,0 +1,272 @@
+                                listCounter++;
+                                listCounter = listCounter >= bingoSquares.length ? 0 : listCounter;
+                                angularVelocityDecrement *= 1.001;
+                                _this.angularVelocityLimit += angularVelocityDecrement;
+                            } else if (currentAngularVelocity.z >= -0.1 && !finalNumber) {
+                                finalNumber = true;
+                                bingoCall = bingoSquares.pop();
+                                if (!bingoCall) {
+                                    print("ALL NUMBERS HAVE BEEN CALLED. START A NEW GAME");
+                                    return;
+                                }
+                                Entities.callEntityServerMethod(_this.entityID, 'addCalledNumber', [bingoCall]);
+                                callNumber = bingoCall.substring(2, bingoCall.length);
+                                Entities.editEntity(bingoNumberTexts[0], {
+                                    locked: false
+                                });
+                                Entities.editEntity(bingoNumberTexts[0], {
+                                    text: bingoCall
+                                });
+                                Entities.editEntity(bingoNumberTexts[0], {
+                                    locked: true
+                                });
+                            } else if (currentAngularVelocity.z >= -0.05) {
+                                if (_this.interval) {
+                                    _this.playSound(BLIP_SOUND);
+                                    Script.clearInterval(_this.interval);
+                                    bingoSquares = [];
+                                    var lightOn = false;
+                                    var blinks = 0;
+                                    _this.interval = Script.setInterval(function() {
+                                        blinks++;
+                                        if (lightOn) {
+                                            _this.lightOff(callNumber);
+                                            lightOn = false;
+                                        } else {
+                                            _this.lightOn(callNumber);
+                                            lightOn = true;
+                                        }
+                                        if (blinks > 6) {
+                                            Script.clearInterval(_this.interval);
+                                            canSpin = true;
+                                        }
+                                    }, LIGHT_BLINK_INTERVAL);
+                                }
+                            }
+                        }, ANGULAR_VELOCITY_CHECK_MS);
+                    }, CHECKING_INTERVAL_DELAY);
+                }
+            }
+        },
+
+        lightOn: function(lightNumber) {
```

Comment:
These should be moved to an entity server script that handles the editing to avoid the lock/unlocks